libraries/ESP8266WiFi/src/ESP8266WiFiGeneric.cpp

```diff
@@ -718,6 +718,48 @@ void wifi_dns_found_callback(const char *name, const ip_addr_t *ipaddr, void *ca
     esp_schedule(); // break delay in hostByName
 }
 
+void wifi_dns_found_callback_async(const char* name, const ip_addr_t* ipaddr, void* callback_arg);
+
+/**
+ * Resolves the given hostname to an IP address asynchronously.
+ * @param aHostname     Name to be resolved
+ * @param aResult       IPAddress structure to store the returned IP address
+ * @param waiting       waiting flag if DNS search is still in progress
+ * @return 1 if aHostname was successfully converted to an IP address,
+ *         else 0
+ */
+
+int ESP8266WiFiGenericClass::hostByNameAsync(const char* aHostname, IPAddress& aResult, uint8_t* waiting) {
+    ip_addr_t addr;
+
+    if (waiting != NULL) {
```

Comment:
`NULL`->`nullptr` everywhere, please.

Reply:
Done in 0093cca821e180a92ba5e6e474faa3f5c5daef7e